### PR TITLE
[fix][broker] Fix namespace backlog quota check with retention.

### LINKED ISSUE #17707: Why pulsar retention quota must exceed configured backlog quota
### Discussed in https://github.com/apache/pulsar/discussions/17697

<div type='discussions-op-text'>

<sup>Originally posted by **mnit016** September 16, 2022</sup>
Hello everyone, 

I'm new to Pulsar, recently I get into a problem when configure the backlog/retention quota for namespace.

The problem:
retention: {time: -1, size: 10M} => then I can set backlog quota as much as I want.
But if retention: {time: 1, size: 10M} => when I set backlog quota exceed 10M, it said "Backlog Quota exceeds configured Retention quota for namespace. Please increase retention quota and retry", I have no idea why.

Could someone please help to explain for me?
Thanks
</div>

### PR DESCRIPTION
**Pull request by Jason918**
Fixes #17707

### Motivation

1. Fix the case mentioned in #17707
2. Fix `checkBacklogQuota` fails when both retention.size and backlogQuota.size is -1. Same as time check.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:
  - testRetentionAndBacklogQuotaCheck

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
bug

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by michaeljmarshall — APPROVED** [review]
LGTM. Thanks @Jason918!

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java [hunk 1]
```diff
@@ -370,17 +370,21 @@ protected CompletableFuture<Optional<TopicPolicies>> getTopicPoliciesAsyncWithRe
     }
 
     protected boolean checkBacklogQuota(BacklogQuota quota, RetentionPolicies retention) {
-        if (retention == null || retention.getRetentionSizeInMB() <= 0 || retention.getRetentionTimeInMinutes() <= 0) {
+        if (retention == null
+                || (retention.getRetentionSizeInMB() <= 0 && retention.getRetentionTimeInMinutes() <= 0)) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:374]: Looks like the added test can't cover this branch?

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:374]: @codelipenghui As mentioned in the issue. The test in L3417 won't fail before this change.

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:374]: And the case of `(retention.getRetentionSizeInMB() <= 0 && retention.getRetentionTimeInMinutes() <= 0)` is the default setting, it's covered by previous unit tests.

**Review by codelipenghui — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**